onmt/encoders/image_encoder.py

```diff
@@ -16,11 +16,12 @@ class ImageEncoder(nn.Module):
         dropout (float): dropout probablity.
     """
 
-    def __init__(self, num_layers, bidirectional, rnn_size, dropout):
+    def __init__(self, num_layers, bidirectional, hidden_size, dropout):
```

Comment:
I think you need to remove all the "image" related changes since I merged the PR fixing all of this + sharding based on number of examples.
Your PR shoudl handle only Speec-2-Text IMO for now.